From 1ed4029766dbf89d26097dd88dcf6d911f23c4c2 Mon Sep 17 00:00:00 2001
From: Moritz Fischer <moritz.fischer@ettus.com>
Date: Mon, 21 Jan 2019 14:09:35 -0800
Subject: [PATCH] ni: zynq: Mender.io support for NI Ettus Research Neon Rev2

This involves adding bootcount in environment, as well as modifying
the default bootconfiguration to run ${mender_setup} first.

Signed-off-by: Moritz Fischer <moritz@ettus.com>

[Adopted to u-boot 2019.07]
Signed-off-by: Joerg Hofrichter <joerg.hofrichter@ni.com>
---
 configs/ni_neon_rev2_defconfig |  5 +++++
 include/configs/ni_neon_rev2.h | 13 ++++++-------
 2 files changed, 11 insertions(+), 7 deletions(-)

diff --git a/configs/ni_neon_rev2_defconfig b/configs/ni_neon_rev2_defconfig
index 6de0e30068..9ec27ff631 100644
--- a/configs/ni_neon_rev2_defconfig
+++ b/configs/ni_neon_rev2_defconfig
@@ -83,4 +83,9 @@ CONFIG_USB_GADGET_PRODUCT_NUM=0x0300
 CONFIG_CI_UDC=y
 CONFIG_USB_GADGET_DOWNLOAD=y
 CONFIG_USB_FUNCTION_THOR=y
+CONFIG_ENV_SIZE=0x20000
+CONFIG_ENV_OFFSET=0x800000
+CONFIG_MMC=y
 CONFIG_ENV_IS_IN_MMC=y
+CONFIG_BOOTCOUNT_LIMIT=y
+CONFIG_BOOTCOUNT_ENV=y
diff --git a/include/configs/ni_neon_rev2.h b/include/configs/ni_neon_rev2.h
index 46b3431951..1ceb3be825 100644
--- a/include/configs/ni_neon_rev2.h
+++ b/include/configs/ni_neon_rev2.h
@@ -11,9 +11,7 @@
 
 #include <configs/zynq-common.h>
 
-#define CONFIG_SYS_MMC_ENV_DEV 0
 #define CONFIG_ENV_SECT_SIZE           CONFIG_ENV_SIZE
-#define CONFIG_ENV_OFFSET              0xE0000
 
 #undef CONFIG_EXTRA_ENV_SETTINGS
 #define CONFIG_EXTRA_ENV_SETTINGS	\
@@ -23,7 +21,7 @@
 	"ec_disable_swsync=0\0"		\
 	"ec_image=lib/firmware/ni/ec-neon-rev2.RW.bin\0"	\
 	"expand_ec_image=setenv ec_image lib/firmware/ni/ec-neon-rev${ec_compat}.RW.bin\0" \
-	"bootargs=root=/dev/mmcblk0p2 rw rootwait uio_pdrv_genirq.of_id=usrp-uio\0" \
+	"bootargs=root=${mender_kernel_root} rw rootwait uio_pdrv_genirq.of_id=usrp-uio\0" \
 	"load_addr=0x2000000\0"		\
 	"fit_size=0x800000\0"           \
 	"fdt_high=0x20000000\0"         \
@@ -34,14 +32,15 @@
 		"echo Loading FIT to RAM via DHCP/TFTP && " \
 		"dhcp ${load_addr} ${tftproot}/${fit_image} && "\
 		"bootm ${load_addr}#conf@zynq-ni-${mboard}.dtb\0" \
-	"sdboot=run swsync_sd; " \
+	"sdboot=run mender_setup; run swsync_sd; " \
 		"echo Copying FIT from SD to RAM... && " \
-		"ext4load mmc 0:2 ${load_addr} ${fit_image} && " \
-		"bootm ${load_addr}#conf@zynq-ni-${mboard}.dtb\0" \
+		"ext4load ${mender_uboot_root} ${load_addr} ${fit_image} && " \
+		"bootm ${load_addr}#conf@zynq-ni-${mboard}.dtb; " \
+		"run mender_try_to_recover\0" \
 	"swsync_sd=if test ${ec_disable_swsync} = 0; then "\
 			"echo Copying MCU FW from SD to RAM... && " \
 			"run expand_ec_image; " \
-			"ext4load mmc 0:2 ${load_addr} ${ec_image} && " \
+			"ext4load ${mender_uboot_root} ${load_addr} ${ec_image} && " \
 			"crosec swsync ${load_addr} ${filesize}; fi;\0" \
 	"swsync_net=if test ${ec_disable_swsync} = 0; then " \
 			"echo Copying MCU FW to RAM via DHCP/TFTP... && " \
-- 
2.17.1

